src/administration/components/AdminPanel.jsx

```diff
@@ -17,160 +12,227 @@ const electron = window.require('electron');
-                                    <td>
-                                        <Button
-                                            variant="danger"
-                                            size="sm"
-                                            onClick={() =>
-                                                this.handleDeleteVolunteer(
-                                                    v.volunteer_id
-                                                )
-                                            }
-                                        >
-                                            {' '}
-                                            Izbriši &nbsp;
-                                            <FaUserMinus />
-                                        </Button>
-                                    </td>
-                                </tr>
-                            );
-                        })}
-                        <tr>
-                            <td colSpan="5">
-                                <div className="border-top my-3"></div>
-                                <h4 className="text-muted">
-                                    <FaPencilAlt /> &nbsp;Unos novog volontera
-                                </h4>
-                                <Form>
-                                    <Form.Group controlId="formBasicFirstName">
-                                        <Form.Label>Ime</Form.Label>
-                                        <input
-                                            type="text"
-                                            name="inputFirstName"
-                                            value={this.state.inputFirstName}
-                                            onChange={this.handleChangeInput}
-                                            className="form-control"
-                                            id="exampleInputFirstName1"
-                                            placeholder="Unesite ime volontera"
-                                        />
-                                    </Form.Group>
-                                    <Form.Group controlId="formBasicLastName">
-                                        <Form.Label>Prezime</Form.Label>
-                                        <Form.Control
-                                            type="text"
-                                            name="inputLastName"
-                                            value={this.state.inputLastName}
-                                            onChange={this.handleChangeInput}
-                                            className="form-control"
-                                            placeholder="Unesite prezime volontera"
-                                        />
-                                    </Form.Group>
+          }
+        this.setState({errorMessage: err});
+        this.setState({[nam]: val});
+        
+    }
+
+    validateField(fieldName, value) {
```

Comment:
If your field is a component, you can validate at that level, instead of procedurally validating each field in the parent.